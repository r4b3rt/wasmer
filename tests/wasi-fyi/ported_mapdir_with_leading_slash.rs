// WASI:
// mapdir: /hamlet:test_fs/hamlet

use std::fs;
use std::io::Read;
use std::path::PathBuf;

fn main() {
    let mut base = PathBuf::from("/hamlet");

    base.push("act3/scene3.txt");

    println!("File exists? {}", base.exists());

    let mut f = fs::File::open(&base).unwrap();
    let mut s = String::new();
    f.read_to_string(&mut s).unwrap();

    println!("{}", s.chars().take(252).collect::<String>());
}
