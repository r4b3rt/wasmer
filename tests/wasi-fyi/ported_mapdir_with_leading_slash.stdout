File exists? true
SCENE III. A room in the castle.

    Enter KING CLAUDIUS, ROSENCRANTZ, and GUILDENSTERN 

KING CLAUDIUS

    I like him not, nor stands it safe with us
    To let his madness range. Therefore prepare you;
    I your commission will forthwith dispatch,
